connectors/gridgo-mysql/src/main/java/io/gridgo/connector/mysql/JdbcProducer.java

```diff
@@ -0,0 +1,65 @@
+package io.gridgo.connector.mysql;
+
+
+import io.gridgo.connector.mysql.support.JdbcOperationException;
+import io.gridgo.connector.support.config.ConnectorContext;
+import static io.gridgo.connector.support.transaction.TransactionConstants.HEADER_CREATE_TRANSACTION;
+import io.gridgo.framework.support.Message;
+import lombok.extern.slf4j.Slf4j;
+import org.jdbi.v3.core.ConnectionFactory;
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.Jdbi;
+import org.joo.promise4j.Promise;
+import org.joo.promise4j.impl.CompletableDeferredObject;
+import org.joo.promise4j.impl.SimpleFailurePromise;
+
+import static io.gridgo.connector.mysql.JdbcConstants.*;
+
+@Slf4j
+class JdbcProducer extends JdbcClient {
+
+    private Jdbi jdbiClient;
+    private String generatedName;
+
+    JdbcProducer(ConnectorContext context, ConnectionFactory connectionFactory) {
+        super(context);
+        this.jdbiClient = Jdbi.create(connectionFactory);
+        this.generatedName = "producer.jdbc";
+    }
+
+    private Message beginTransaction(Message msg, Handle handle, ConnectorContext context) {
+        handle.begin();
+        JdbcTransaction jdbcTransaction = new JdbcTransaction(handle, context);
+        return Message.ofAny(jdbcTransaction);
+    }
+
+    protected Promise<Message, Exception> _call(Message request, CompletableDeferredObject<Message, Exception> deferred, boolean isRPC) {
+        if (deferred == null) {
+            return null;
+        }
+        var operation = request.headers().getString(OPERATION);
+        if (HEADER_CREATE_TRANSACTION.equals(operation)) {
+            var result = beginTransaction(request, jdbiClient.open(), this.getContext());
+            ack(deferred, result);
+            return deferred.promise();
+        }
+        var handler = operationsMap.get(operation);
+        if (handler == null) {
+            return new SimpleFailurePromise<>(new JdbcOperationException());
+        }
+        try (Handle handle = jdbiClient.open()) {
```

Comment:
should use an execution strategy here

```
var runnable = () -> {
    // move the whole try-catch block here
}
getConnectorContext().getProducerExecutionStrategy()
                     .ifPresentOrElse(strategy -> strategy.execute(runnable), runnable);
```

or better, use a default execution strategy (`ExecutorExecutionStrategy`) if it's not passed inside `connectorContext`